src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+import net.minecraftforge.fmp.client.multipart.ICustomHighlightPart;
+import net.minecraftforge.fmp.client.multipart.MultipartStateMapper;
+import net.minecraftforge.fmp.microblock.IMicroblockContainerTile;
+import net.minecraftforge.fmp.microblock.MicroblockContainer;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.Multipart;
+import net.minecraftforge.fmp.multipart.MultipartContainer;
+import net.minecraftforge.fmp.multipart.MultipartRegistry;
+import net.minecraftforge.fmp.multipart.PartState;
+import net.minecraftforge.fmp.raytrace.RayTraceUtils;
+
+/**
+ * An {@link Block} that allows your block to act as a microblock container.<br/>
+ * Extend this class if you want your block to be able to have microblocks, but not any other kinds of multiparts.
+ * Otherwise, extend {@link Multipart} or make a custom implementation of {@link IMultipart}.<br/>
+ * All the overriden methods have a "default" counterpart that allows you to handle interactions with your block if the
+ * player isn't interacting with a multipart.<br/>
+ * Extend {@link TileCoverable} or implement {@link IMicroblockContainerTile} and return it in
+ * {@link BlockCoverable#createNewTileEntity(World, int)} if you want a custom tile entity for your block.
+ */
+public class BlockCoverable extends Block implements ITileEntityProvider
+{
+    
+    private AxisAlignedBB bounds = FULL_BLOCK_AABB;
+
+    public BlockCoverable(Material material)
+    {
+        super(material);
+        MinecraftForge.EVENT_BUS.register(this);
+    }
+
+    @Override
+    public TileEntity createNewTileEntity(World worldIn, int meta)
+    {
+        return createNewTileEntityDefault(worldIn, meta);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends TileEntity & IMicroblockContainerTile> T createNewTileEntityDefault(World worldIn, int meta)
+    {
+        return (T) new TileCoverable();
+    }
+
+    protected IMicroblockContainerTile getMicroblockTile(IBlockAccess world, BlockPos pos)
+    {
+        TileEntity tile = world.getTileEntity(pos);
+        return tile instanceof IMicroblockContainerTile ? (IMicroblockContainerTile) tile : null;
+    }
+
+    @Override
+    public final RayTraceResult collisionRayTrace(IBlockState state, World world, BlockPos pos, Vec3d start, Vec3d end)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        RayTraceResult result = tile != null ? tile.getMicroblockContainer().getPartContainer().collisionRayTrace(start, end)
+                : null;
+        RayTraceResult hit = collisionRayTraceDefault(state, world, pos, start, end);
+        if (result == null || (hit != null && hit.hitVec.squareDistanceTo(start) < result.hitVec.squareDistanceTo(start)))
+        {
+            bounds = getSelectedBoundingBoxDefault(state, world, pos).offset(-pos.getX(), -pos.getY(), -pos.getZ());
+            return hit;
+        }
+        bounds = result.boxHit;
+        return result;
```

Comment:
This was the only way to handle something like this, tbh. When the selection box is retrieved, no external information is provided, so you have to save it when you raytrace it and provide it later. Not the cleanest solution, but I'd blame vanilla for it.